contract/src/main.rs

```diff
@@ -16,193 +19,139 @@ use types::{
     account::AccountHash,
     bytesrepr::{FromBytes, ToBytes},
     contracts::{EntryPoint, EntryPointAccess, EntryPointType, EntryPoints},
-    runtime_args, CLType, CLTyped, Group, Key, Parameter, PublicKey, RuntimeArgs, URef, U128, U256,
-    U512,
+    runtime_args, ApiError, CLType, CLTyped, CLValue, Group, Key, Parameter, PublicKey,
+    RuntimeArgs, URef, U128, U256, U512,
 };
 
+// Create a new storage value under dict_key in the dictionary accessed by uref
 #[no_mangle]
-pub extern "C" fn store_bool() {
-    read_and_store::<bool>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_i32() {
-    read_and_store::<i32>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_i64() {
-    read_and_store::<i64>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u8() {
-    read_and_store::<u8>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u32() {
-    read_and_store::<u32>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u64() {
-    read_and_store::<u64>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u128() {
-    read_and_store::<U128>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u256() {
-    read_and_store::<U256>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_u512() {
-    read_and_store::<U512>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_string() {
-    read_and_store::<String>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_key() {
-    read_and_store::<Key>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_list_of_bytes() {
-    read_and_store::<Vec<u8>>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_public_key() {
-    read_and_store::<PublicKey>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_option() {
-    read_and_store::<Option<String>>();
-}
-
-#[no_mangle]
-pub extern "C" fn store_result() {
-    read_and_store::<Result<String, String>>();
+fn create() {
+    let name: String = runtime::get_named_arg("uref");
+    let key: String = runtime::get_named_arg("dict_key");
+    let value: Option<String> = runtime::get_named_arg("dict_value");
```

Comment:
Creating a new value should not be `Option` -> Obfuscate `Option` data management